### PR TITLE
fix: add health check interval to reset service if an issue is detected

### PR DESCRIPTION
**Pull request by MikeLuDev**
## Changes

- Adds health check interval to `GmOrderbooksService` to reset everything if an update hasn't occurred after a certain number of threshold seconds

## Checklist

- [ ] UAT
- [ ] Passes final QA checks

### CODE REVIEW

File: src/marketplace/services/GmOrderbookService.ts [hunk 1]
```diff
@@ -60,10 +63,14 @@ export class GmOrderbookService {
       250
     );
 
-    this.GmEventService.setEventHandler(this.handleMarketplaceEvent);
+    this.gmEventService.setEventHandler(this.handleMarketplaceEvent);
 
     await this.loadInitialOrders();
 
+    this.healthcheckTimer = setInterval(() => {
+      this.checkServiceHealth();
```
Comment (staratlasjim) [inline src/marketplace/services/GmOrderbookService.ts:71]: @MikeLuDev  - I suggest adding a flag that skips the reset if the code is awaiting the reset finishing. Basically, what if it takes more than 5 seconds to get the reset/reload of data to finish, could it accidentally re-trigger the kick off?

something like this pseudo code
```
this.healthcheckTimer = ... => {
 if(this.isReloading) return;
 this.isReloading = true;
 this.checkServiceHealth().then(()=> this.isReloading = false );
}
```

**Review by staratlasjim — CHANGES_REQUESTED** [review]
@MikeLuDev  - made two small suggestions.

File: src/marketplace/services/GmOrderbookService.ts (hunk 1 above)
Comment (MikeLuDev) [reply src/marketplace/services/GmOrderbookService.ts:71]: Good point :+1:

File: src/marketplace/services/GmOrderbookService.ts [hunk 2]
```diff
@@ -74,15 +81,69 @@ export class GmOrderbookService {
...
 
-  public addOnEventHandler(
-    eventHandler: GmEventHandler
-  ): void {
+  protected async checkServiceHealth(): Promise<void> {
+    const now = this.getNow();
+    const secondsSinceLastEvent = now - this.lastEventTimestamp;
+    const isUnhealthy = secondsSinceLastEvent >= this.healthcheckThreshold;
+
+    if (isUnhealthy) {
```
[the remark this answers is not in the document]
Comment (MikeLuDev) [reply src/marketplace/services/GmOrderbookService.ts:89]: Will do :+1:

**Pull request merged**